upup/pkg/kutil/rollingupdate_cluster_validate_drain.go

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```

Comment:
The problem is the structs.  I have modified the data structures.  Yes, I would do that, but there is less risk to the code with a copy-pasta.  ... Y/N?